src/main/java/persistence/sql/dml/DeleteQueryBuilder.java

```diff
@@ -1,12 +1,20 @@
 package persistence.sql.dml;
 
-import persistence.sql.Metadata;
+import persistence.sql.entity.EntityColumns;
+import persistence.sql.entity.EntityTable;
 
 public class DeleteQueryBuilder {
-    public String delete(Class<?> clazz, Object idValue) {
-        Metadata metadata = new Metadata(clazz);
-        String tableName = metadata.getTableName();
-        String idField = metadata.getIdFieldName();
+    private final EntityTable entityTable;
+    private final EntityColumns entityColumns;
+
+    public DeleteQueryBuilder(EntityTable entityTable, EntityColumns entityColumns) {
+        this.entityTable = entityTable;
+        this.entityColumns = entityColumns;
+    }
+
+    public String delete(Object idValue) {
+        String tableName = entityTable.getTableName();
+        String idField = entityColumns.getIdFieldName();
```

Comment:
메서드 파라미터로 엔티티와 관련된 정보를 받도록 설정하면 
`QueryBuilder`를 클래스마다 생성하지 않아 메모리적인 이점이 있을 수 있을 것 같아요 :)

```suggestion
    public String delete(EntityTable entityTable, EntityColumn idColumn, Object idValue) {
        String tableName = entityTable.getTableName();
        String idField = entityColumns.getIdFieldName();
```

Reply:
쿼리를 생성하는 쿼리빌더에서 굳이 EntityTable과 EntityColumn을 들고있을필요가 없었네요. 좋은 의견 감사합니다.